src/Nethermind/Nethermind.Serialization.Rlp/Rlp.cs

```diff
@@ -422,8 +422,8 @@ public static int Encode(Span<byte> buffer, int position, Hash256 hash)
                 ThrowArgumentOutOfRangeException();
             }
 
-            Unsafe.Add(ref MemoryMarshal.GetReference(buffer), position) = 160;
-            Unsafe.As<byte, ValueHash256>(ref Unsafe.Add(ref MemoryMarshal.GetReference(buffer), position + 1)) = hash.ValueHash256;
+            Unsafe.Add(ref MemoryMarshal.GetReference(buffer), (nuint)position) = 160;
+            Unsafe.As<byte, ValueHash256>(ref Unsafe.Add(ref MemoryMarshal.GetReference(buffer), (nuint)position + 1)) = hash.ValueHash256;
```

Comment:
why `nuint`? We only support 64bit systems already.

Reply:
Intent; the cast is to avoid some additional movsx that occur with `int`; could cast to `ulong` but the reason it to be unsigned and machine word size so `nuint` makes more sense?